### ISSUE TITLE
bug: doc CI build failed

### ISSUE THREAD

**Issue by wusyong** [tauri-apps/wry#1117]

**Comment by wusyong** [comment]
I saw this doc script is added by @rhysd. Could you help us understand why it is failing?

**Comment by rhysd** [comment]
Extracted from [this job](https://github.com/tauri-apps/wry/actions/runs/7125970240/job/19437087907?pr=1109):

```
thread 'rustc' panicked at compiler/rustc_infer/src/infer/generalize.rs:28:40:
called `Result::unwrap_err()` on an `Ok` value: ?11t
stack backtrace:
   0:     0x7f88ae18b45c - std::backtrace_rs::backtrace::libunwind::trace::h815354fef816bc86
                               at /rustc/503e129328080e924c0ddfca6abf4c2812580102/library/std/src/../../backtrace/src/backtrace/libunwind.rs:104:5
   1:     0x7f88ae18b45c - std::backtrace_rs::backtrace::trace_unsynchronized::h8ee509da58073872
                               at /rustc/503e129328080e924c0ddfca6abf4c2812580102/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x7f88ae18b45c - std::sys_common::backtrace::_print_fmt::ha7526b8479823d1a
                               at /rustc/503e129328080e924c0ddfca6abf4c2812580102/library/std/src/sys_common/backtrace.rs:68:5
   3:     0x7f88ae18b45c - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::haf74540a80629043
                               at /rustc/503e129328080e924c0ddfca6abf4c2812580102/library/std/src/sys_common/backtrace.rs:44:22
   4:     0x7f88ae1de4a0 - core::fmt::rt::Argument::fmt::hc68f5fa1b36668fc
                               at /rustc/503e129328080e924c0ddfca6abf4c2812580102/library/core/src/fmt/rt.rs:142:9
   5:     0x7f88ae1de4a0 - core::fmt::write::h4274fed398e9428e
                               at /rustc/503e129328080e924c0ddfca6abf4c2812580102/library/core/src/fmt/mod.rs:1120:17
   6:     0x7f88ae17f34f - std::io::Write::write_fmt::hc6c4a8568e7df724
                               at /rustc/503e129328080e924c0ddfca6abf4c2812580102/library/std/src/io/mod.rs:1810:15
   7:     0x7f88ae18b244 - std::sys_common::backtrace::_print::h2e09ef8543bf3bd8
                               at /rustc/503e129328080e924c0ddfca6abf4c2812580102/library/std/src/sys_common/backtrace.rs:47:5
   8:     0x7f88ae18b244 - std::sys_common::backtrace::print::h9a43f66a9d093046
                               at /rustc/503e129328080e924c0ddfca6abf4c2812580102/library/std/src/sys_common/backtrace.rs:34:9
   9:     0x7f88ae18ded7 - std::panicking::default_hook::{{closure}}::h9c00a7d25a3e4c19
  10:     0x7f88ae18dc3f - std::panicking::default_hook::h1b24ab04fe68d03a
                               at /rustc/503e129328080e924c0ddfca6abf4c2812580102/library/std/src/panicking.rs:292:9
  11:     0x7f88b0ef82a0 - std[ce0584ef05f9a8b4]::panicking::update_hook::<alloc[9ee1258aa4f01231]::boxed::Box<rustc_driver_impl[42a0b4a1f0a933f6]::install_ice_hook::{closure#0}>>::{closure#0}
  12:     0x7f88ae18e618 - <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call::h4054d74742b8ea41
                               at /rustc/503e129328080e924c0ddfca6abf4c2812580102/library/alloc/src/boxed.rs:2029:9
  13:     0x7f88ae18e618 - std::panicking::rust_panic_with_hook::hbafd4f0baa9fdb9e
                               at /rustc/503e129328080e924c0ddfca6abf4c2812580102/library/std/src/panicking.rs:783:13
  14:     0x7f88ae18e36e - std::panicking::begin_panic_handler::{{closure}}::h4c976a34ccffa9e2
                               at /rustc/503e129328080e924c0ddfca6abf4c2812580102/library/std/src/panicking.rs:657:13
  15:     0x7f88ae18b926 - std::sys_common::backtrace::__rust_end_short_backtrace::h675039fe30aa0955
                               at /rustc/503e129328080e924c0ddfca6abf4c2812580102/library/std/src/sys_common/backtrace.rs:171:18
  16:     0x7f88ae18e0d2 - rust_begin_unwind
                               at /rustc/503e129328080e924c0ddfca6abf4c2812580102/library/std/src/panicking.rs:645:5
  17:     0x7f88ae1dab65 - core::panicking::panic_fmt::h32ca55ac70be178d
                               at /rustc/503e129328080e924c0ddfca6abf4c2812580102/library/core/src/panicking.rs:72:14
  18:     0x7f88ae1db273 - core::result::unwrap_failed::h179c95b92f11ec78
                               at /rustc/503e129328080e924c0ddfca6abf4c2812580102/library/core/src/result.rs:1649:5
  19:     0x7f88b225b2c8 - <rustc_infer[8d75f53857709946]::infer::equate::Equate as rustc_middle[3b7dc7f1ac8ce2bc]::ty::relate::TypeRelation>::tys
  20:     0x7f88b28fc9f4 - <rustc_trait_selection[cc4a1b9dcbaa9312]::traits::select::SelectionContext>::candidate_from_obligation_no_cache
  21:     0x7f88af32f0d8 - <rustc_trait_selection[cc4a1b9dcbaa9312]::traits::select::SelectionContext>::poly_select
  22:     0x7f88b24b7645 - rustc_trait_selection[cc4a1b9dcbaa9312]::traits::project::opt_normalize_projection_type
  23:     0x7f88b244b42a - rustc_trait_selection[cc4a1b9dcbaa9312]::traits::project::poly_project_and_unify_type
  24:     0x7f88b2f482c6 - <rustc_trait_selection[cc4a1b9dcbaa9312]::traits::select::SelectionContext>::evaluate_predicate_recursively
  25:     0x7f88b2f47f8b - <rustc_trait_selection[cc4a1b9dcbaa9312]::traits::select::SelectionContext>::evaluate_predicates_recursively::<alloc[9ee1258aa4f01231]::vec::Vec<rustc_infer[8d75f53857709946]::traits::Obligation<rustc_middle[3b7dc7f1ac8ce2bc]::ty::Predicate>>>
  26:     0x7f88b2f484b3 - <rustc_trait_selection[cc4a1b9dcbaa9312]::traits::select::SelectionContext>::evaluate_predicate_recursively
  27:     0x7f88b2907a46 - <rustc_trait_selection[cc4a1b9dcbaa9312]::traits::select::SelectionContext>::candidate_from_obligation_no_cache
  28:     0x7f88af32f0d8 - <rustc_trait_selection[cc4a1b9dcbaa9312]::traits::select::SelectionContext>::poly_select
  29:     0x7f88b24b7645 - rustc_trait_selection[cc4a1b9dcbaa9312]::traits::project::opt_normalize_projection_type
  30:     0x7f88b24b55f8 - <rustc_trait_selection[cc4a1b9dcbaa9312]::traits::project::AssocTypeNormalizer as rustc_type_ir[ee20701d43e568e4]::fold::TypeFolder<rustc_middle[3b7dc7f1ac8ce2bc]::ty::context::TyCtxt>>::fold_ty
  31:     0x7f88b24b4608 - <rustc_trait_selection[cc4a1b9dcbaa9312]::traits::project::AssocTypeNormalizer as rustc_type_ir[ee20701d43e568e4]::fold::TypeFolder<rustc_middle[3b7dc7f1ac8ce2bc]::ty::context::TyCtxt>>::fold_predicate
  32:     0x7f88b2c4b34a - <rustc_trait_selection[cc4a1b9dcbaa9312]::traits::select::SelectionContext>::confirm_candidate
  33:     0x7f88b2c3e611 - <rustc_trait_selection[cc4a1b9dcbaa9312]::traits::select::SelectionContext>::evaluate_candidate
  34:     0x7f88b2901584 - <rustc_trait_selection[cc4a1b9dcbaa9312]::traits::select::SelectionContext>::candidate_from_obligation_no_cache
  35:     0x7f88afedf04b - <rustc_trait_selection[cc4a1b9dcbaa9312]::traits::select::SelectionContext>::candidate_from_obligation
  36:     0x7f88b2be7425 - <rustc_trait_selection[cc4a1b9dcbaa9312]::traits::select::SelectionContext>::evaluate_trait_predicate_recursively
  37:     0x7f88b2c3e8fa - <rustc_trait_selection[cc4a1b9dcbaa9312]::traits::select::SelectionContext>::evaluate_candidate
  38:     0x7f88b2be745c - <rustc_trait_selection[cc4a1b9dcbaa9312]::traits::select::SelectionContext>::evaluate_trait_predicate_recursively
  39:     0x7f88b2f4258d - <rustc_infer[8d75f53857709946]::infer::InferCtxt>::probe::<core[a6856156d288c90]::result::Result<rustc_middle[3b7dc7f1ac8ce2bc]::traits::select::EvaluationResult, rustc_middle[3b7dc7f1ac8ce2bc]::traits::select::OverflowError>, <rustc_trait_selection[cc4a1b9dcbaa9312]::traits::select::SelectionContext>::evaluation_probe<<rustc_trait_selection[cc4a1b9dcbaa9312]::traits::select::SelectionContext>::evaluate_root_obligation::{closure#0}>::{closure#0}>
  40:     0x7f88b240e890 - rustc_traits[6896ab33d75d3fc9]::evaluate_obligation::evaluate_obligation
  41:     0x7f88b240e260 - rustc_query_impl[d233ef11b2077f1a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[d233ef11b2077f1a]::query_impl::evaluate_obligation::dynamic_query::{closure#2}::{closure#0}, rustc_middle[3b7dc7f1ac8ce2bc]::query::erase::Erased<[u8; 2usize]>>
  42:     0x7f88b240d810 - rustc_query_system[50084ec5a8fda77a]::query::plumbing::try_execute_query::<rustc_query_impl[d233ef11b2077f1a]::DynamicConfig<rustc_query_system[50084ec5a8fda77a]::query::caches::DefaultCache<rustc_type_ir[ee20701d43e568e4]::canonical::Canonical<rustc_middle[3b7dc7f1ac8ce2bc]::ty::context::TyCtxt, rustc_middle[3b7dc7f1ac8ce2bc]::ty::ParamEnvAnd<rustc_middle[3b7dc7f1ac8ce2bc]::ty::Predicate>>, rustc_middle[3b7dc7f1ac8ce2bc]::query::erase::Erased<[u8; 2usize]>>, false, false, false>, rustc_query_impl[d233ef11b2077f1a]::plumbing::QueryCtxt, false>
  43:     0x7f88b240d495 - rustc_query_impl[d233ef11b2077f1a]::query_impl::evaluate_obligation::get_query_non_incr::__rust_end_short_backtrace
  44:     0x7f88af392c5d - <rustc_infer[8d75f53857709946]::infer::InferCtxt as rustc_trait_selection[cc4a1b9dcbaa9312]::traits::query::evaluate_obligation::InferCtxtExt>::evaluate_obligation
  45:     0x5602743f750a - rustdoc[608054bee38[698](https://github.com/tauri-apps/wry/actions/runs/7125970240/job/19437087907?pr=1109#step:5:699)30]::clean::utils::get_auto_trait_and_blanket_impls
  46:     0x560274504453 - <rustdoc[608054bee3869830]::passes::collect_trait_impls::SyntheticImplCollector as rustdoc[608054bee3869830]::visit::DocVisitor>::visit_item
  47:     0x5602745047ca - <rustdoc[608054bee3869830]::passes::collect_trait_impls::SyntheticImplCollector as rustdoc[608054bee3869830]::visit::DocVisitor>::visit_item
  48:     0x5602745047ca - <rustdoc[608054bee3869830]::passes::collect_trait_impls::SyntheticImplCollector as rustdoc[608054bee3869830]::visit::DocVisitor>::visit_item
  49:     0x5602745047ca - <rustdoc[608054bee3869830]::passes::collect_trait_impls::SyntheticImplCollector as rustdoc[608054bee3869830]::visit::DocVisitor>::visit_item
  50:     0x5602745010cc - rustdoc[608054bee3869830]::passes::collect_trait_impls::collect_trait_impls
  51:     0x56027442b3e8 - rustdoc[608054bee3869830]::core::run_global_ctxt
  52:     0x560274531dda - rustdoc[608054bee3869830]::main_args::{closure#1}::{closure#0}::{closure#0}
  53:     0x56027433118e - rustc_interface[8941d9[703](https://github.com/tauri-apps/wry/actions/runs/7125970240/job/19437087907?pr=1109#step:5:704)01189ef]::interface::run_compiler::<core[a6856156d288c90]::result::Result<(), rustc_span[79cd123206d5c77b]::ErrorGuaranteed>, rustdoc[608054bee3869830]::main_args::{closure#1}>::{closure#0}
  54:     0x560274301cb2 - std[ce0584ef05f9a8b4]::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface[8941d970301189ef]::util::run_in_thread_with_globals<rustc_interface[8941d970301189ef]::util::run_in_thread_pool_with_globals<rustc_interface[8941d970301189ef]::interface::run_compiler<core[a6856156d288c90]::result::Result<(), rustc_span[79cd123206d5c77b]::ErrorGuaranteed>, rustdoc[608054bee3869830]::main_args::{closure#1}>::{closure#0}, core[a6856156d288c90]::result::Result<(), rustc_span[79cd123206d5c77b]::ErrorGuaranteed>>::{closure#0}, core[a6856156d288c90]::result::Result<(), rustc_span[79cd123206d5c77b]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[a6856156d288c90]::result::Result<(), rustc_span[79cd123206d5c77b]::ErrorGuaranteed>>
  55:     0x56027433bf6e - <<std[ce0584ef05f9a8b4]::thread::Builder>::spawn_unchecked_<rustc_interface[8941d970301189ef]::util::run_in_thread_with_globals<rustc_interface[8941d970301189ef]::util::run_in_thread_pool_with_globals<rustc_interface[8941d970301189ef]::interface::run_compiler<core[a6856156d288c90]::result::Result<(), rustc_span[79cd123206d5c77b]::ErrorGuaranteed>, rustdoc[608054bee3869830]::main_args::{closure#1}>::{closure#0}, core[a6856156d288c90]::result::Result<(), rustc_span[79cd123206d5c77b]::ErrorGuaranteed>>::{closure#0}, core[a6856156d288c90]::result::Result<(), rustc_span[79cd123206d5c77b]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[a6856156d288c90]::result::Result<(), rustc_span[79cd123206d5c77b]::ErrorGuaranteed>>::{closure#1} as core[a6856156d288c90]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  56:     0x7f88ae1984a5 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h2d19f142671865ec
                               at /rustc/503e129328080e924c0ddfca6abf4c2812580102/library/alloc/src/boxed.rs:2015:9
  57:     0x7f88ae1984a5 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h[710](https://github.com/tauri-apps/wry/actions/runs/7125970240/job/19437087907?pr=1109#step:5:711)d713fa9cb[712](https://github.com/tauri-apps/wry/actions/runs/7125970240/job/19437087907?pr=1109#step:5:713)7
                               at /rustc/503e129328080e924c0ddfca6abf4c2812580102/library/alloc/src/boxed.rs:2015:9
  58:     0x7f88ae1984a5 - std::sys::unix::thread::Thread::new::thread_start::haaff457ad882d05b
                               at /rustc/503e129328080e924c0ddfca6abf4c2812580102/library/std/src/sys/unix/thread.rs:108:17
  59:     0x7f88adc94ac3 - <unknown>
  60:     0x7f88add26a40 - <unknown>
  61:                0x0 - <unknown>

error: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-rustdoc&template=ice.md

note: please attach the file at `/home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-util-0.3.29/rustc-ice-2023-12-08T04_21_34-4922.txt` to your bug report

note: compiler flags: --crate-type lib

note: some of the compiler flags provided by cargo are hidden

query stack during panic:
#0 [evaluate_obligation] evaluating trait selection obligation `stream::stream::FlatMapUnordered<^1_1, ^1_2, ^1_0>: futures_core::stream::Stream`
end of query stack
error: could not document `futures-util`
```

Looks like the bug reported by this issue: https://github.com/rust-lang/rust/issues/118760

This seems a `rustc` compiler bug. Since we use a nightly compiler, sometimes we encounter such ICEs.

**Comment by rhysd** [comment]
The issue seemed to be fixed by this PR 3 days ago: https://github.com/rust-lang/rust/pull/118710

I guess now re-running the job would pass since the fix is included in the latest nightly compiler.

**Comment by pewsheen** [comment]
> The issue seemed to be fixed by this PR 3 days ago: [rust-lang/rust#118710](https://github.com/rust-lang/rust/pull/118710)
> 
> I guess now re-running the job would pass since the fix is included in the latest nightly compiler.

Wow it passed now
